skimage/segmentation/slic_superpixels.py

```diff
@@ -1,17 +1,101 @@
+import warnings
 import collections as coll
 import numpy as np
 from scipy import ndimage as ndi
+from scipy.spatial.distance import pdist, squareform
 
+from ._slic import (_slic_cython, _enforce_label_connectivity_cython)
 from ..util import img_as_float, regular_grid
-from ..segmentation._slic import (_slic_cython,
-                                  _enforce_label_connectivity_cython)
 from ..color import rgb2lab
 
 
+def _get_mask_centroids(mask, n_centroids, spacing=None):
+    """Find regularly spaced centroids on a mask.
+
+    Parameters
+    ----------
+    mask : 3D ndarray
+        The mask where the centroids must be positioned.
+    n_centroids : int
+        The number of centroids to be returned.
+    spacing : sequence of same, optional,
+        Spacing of elements along each dimension.
+
+    Returns
+    -------
+    centroids : 2D ndarray
+        The coordinates of the centroids with shape (n_centroids, 3).
+    steps : 1D ndarray
+        The approximate distance between two seeds in all dimensions.
+
+    """
+    if spacing is None:
+        spacing = np.ones(3)
+
+    coord = np.asarray(np.nonzero(mask))
+    bbox = coord.min(-1), coord.max(-1) + 1
+    roi = mask[bbox[0][0]: bbox[1][0],
+               bbox[0][1]: bbox[1][1],
+               bbox[0][2]: bbox[1][2]].copy()
+
+    centroids = np.repeat([bbox[0]], n_centroids, axis=0)
+
+    for idx in range(n_centroids):
+        dist_map = ndi.distance_transform_edt(roi, sampling=spacing)
```

Comment:
I did a quick profiling of the gallery example, and this line is taking 75% of the total CPU time. Would it be OK to replace it by ``distance_transform_cdt`` which is slightly less accurate but would still be OK for the algorithm (I think)? On the gallery example the result looks as good, and it allows to gain a 2.5 factor (from 2.6 s to 1s on my box).

Reply:
or we could use euclidean when spacing is not None, and cdt otherwise for better performance.

Reply:
Good suggestion, using `distance_transform_cdt` should not affect the quality of the result.